src/Panel/Controller/Dialog/FilesPickerDialogController.php

```diff
@@ -65,6 +66,27 @@ public function find(string $id): File|null
 		return $this->kirby->file($id, $this->model);
 	}
 
+	/**
+	 * Returns the item data for a file
+	 * @param \Kirby\Cms\File $model
+	 */
+	public function item(ModelWithContent $model): array
+	{
+		$props = (new FileItem(
+			file:               $model,
+			dragTextIsAbsolute: $model->parent()->is($this->model),
+			image:              $this->image,
+			info:               $this->info,
+			layout:             $this->layout,
+			text:               $this->text
+		))->props();
+
+		return [
+			...$props,
+			'id' => $props['uuid'] ?? $props['id'],
+		];
```

Comment:
This is quite ugly, but I didn't know a better solution within scope. The problem is that at the latest batch mode in the collection will pass around only the `id`. So we need to have the UUID if available in the `id` field.

One alternative (untested) could be that we remove UUIDs completely from the communication between backend and frontend here, but always use the regular IDs. And only when storing to content we use the UUID.